joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/parser/RegexLiteralHandling.scala

```diff
@@ -69,10 +69,19 @@ trait RegexLiteralHandling { this: RubyLexerBase =>
     * may not start with a WS. E.g. `puts /x/` is valid, but `puts / x/` is not.
     */
   private def isInCommandArgumentPosition: Boolean = {
-    val previousNonWsIsIdentifier = previousNonWsTokenTypeOrEOF() == LOCAL_VARIABLE_IDENTIFIER
-    val previousIsWs              = previousTokenTypeOrEOF() == WS
-    val nextCharIsWs              = _input.LA(1) == ' '
+    val previousNonWsIsIdentifier =
+      previousNonWsTokenTypeOrEOF() == LOCAL_VARIABLE_IDENTIFIER || isControlStructureStart
+    val previousIsWs = previousTokenTypeOrEOF() == WS
+    val nextCharIsWs = _input.LA(1) == ' '
     previousNonWsIsIdentifier && previousIsWs && !nextCharIsWs
   }
 
+  private def isControlStructureStart: Boolean = {
+    previousNonWsTokenTypeOrEOF() == IF
+    // TODO: Determine which of the below control structures are also valid for regex use
```

Comment:
I think we should address this condition in this PR. Maybe do a quick check in an interpreter, then we avoid having to revisit this